### ISSUE TITLE
Select field causes errors in prod env only.

### ISSUE THREAD

**Issue by milosa** [bolt/core#1798]
Using a content type with a select field causes errors when using `APP_ENV=prod` but works fine when using `APP_ENV=dev`.

The problem is that I can access the 'new' page for the content type only once in `prod`. When I try to save the item, I get these errors.

```
php.CRITICAL: Uncaught Error: Argument 1 passed to Bolt\Entity\Content::setStatus() must be of the type string, null given, called in /var/www/vendor/bolt/core/src/Entity/Content.php on line 189 {"exception":"[object] (TypeError(code: 0): Argument 1 passed to Bolt\\Entity\\Content::setStatus() must be of the type string, null given, called in /var/www/vendor/bolt/core/src/Entity/Content.php on line 189 at /var/www/vendor/bolt/core/src/Entity/Content.php:348)"} []
```
And

```
request.CRITICAL: Uncaught PHP Exception Doctrine\ORM\ORMInvalidArgumentException: "Detached entity Array cannot be removed" at /var/www/vendor/doctrine/orm/lib/Doctrine/ORM/ORMInvalidArgumentException.php line 177 {"exception":"[object] (Doctrine\\ORM\\ORMInvalidArgumentException(code: 0): Detached entity Array cannot be removed at /var/www/vendor/doctrine/orm/lib/Doctrine/ORM/ORMInvalidArgumentException.php:177)"} []
```

Then, when I try to re-open the new page, I get this error:

```
php.CRITICAL: Uncaught Error: Argument 1 passed to Bolt\Twig\ContentExtension::selectOptions() must be an instance of Bolt\Entity\Field\SelectField, instance of Bolt\Entity\Field given, called in /var/www/var/cache/prod/twig/5c/5c3c4c0128691277917e2f3e1fa036ae483d1ca516e7a4d91845c4a79d5145ca.php on line 44 {"exception":"[object] (TypeError(code: 0): Argument 1 passed to Bolt\\Twig\\ContentExtension::selectOptions() must be an instance of Bolt\\Entity\\Field\\SelectField, instance of Bolt\\Entity\\Field given, called in /var/www/var/cache/prod/twig/5c/5c3c4c0128691277917e2f3e1fa036ae483d1ca516e7a4d91845c4a79d5145ca.php on line 44 at /var/www/vendor/bolt/core/src/Twig/ContentExtension.php:505)"} []
```

When I change the select field type to a text type, none of these problems occur. 

The select field looked like this:

```
        field_name:
            label: Some label
            type: select
            values: [ foo, bar ]
            multiple: false
            default: [ foo ]
```

Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | 4.0 RC39
| Install type                   |  Composer
| PHP version                 | 7.4.9
| Web server                  | Apache

**Comment by bobdenotter** [comment]
@milosa Hmm, is this on SQLite or MySQL?

**Comment by milosa** [comment]
SQLite

**Comment by bobdenotter** [comment]
Ok.. When you say: 

> When I try to save the item, I get these errors.

Where do you get these errors? In the browser?

**Comment by milosa** [comment]
I get them from the prod.log file. The browser just gives a 500 error.

**Comment by bobdenotter** [comment]
Ok, i'm gonna see if i can reproduce! :-)

**Comment by bobdenotter** [comment]
@milosa This should be fixed in #1800
